Storage/Storage/Tools/StorageType+Deletions.swift

```diff
@@ -54,4 +54,29 @@ public extension StorageType {
             deleteObject(category)
         }
     }
+
+    /// Deletes all of the stored Product Tags that don't have an active product relationship
+    ///
+    func deleteUnusedProductTags(siteID: Int64) {
+        let tagsWithNoAssociatedProducts = loadProductTags(siteID: siteID).filter { tag in
+            guard let products = tag.products else {
+                return true
+            }
+            return products.isEmpty
+        }
+        tagsWithNoAssociatedProducts.forEach { tag in
+            deleteObject(tag)
+        }
+    }
+
+    /// Deletes all the stored Product Tags with the specified IDs
+    ///
+    func deleteProductTags(siteID: Int64, ids: [Int64]) {
+        let tagsWithSpecifiedIDs = loadProductTags(siteID: siteID).filter { tag in
```

Comment:
nit: probably can make this one line like `loadProductTags(siteID: siteID).filter { ids.contains($0.tagID) }`?